modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/TimeStampTest.java

```diff
@@ -0,0 +1,189 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+#
+# Integration tests (client-side) of timestamp
+# Requires the full stack to be running: service, DB, Cassandra, log uploader
+#
+*/
+
+package com.intuit.wasabi.tests.service;
+
+import com.intuit.wasabi.tests.library.TestBase;
+import com.intuit.wasabi.tests.library.util.TestUtils;
+import com.intuit.wasabi.tests.model.*;
+import com.intuit.wasabi.tests.model.analytics.AnalyticsParameters;
+import com.intuit.wasabi.tests.model.factory.EventFactory;
+import com.intuit.wasabi.tests.model.factory.ExperimentFactory;
+import com.intuit.wasabi.tests.service.segmentation.BatchRuleTest;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.intuit.wasabi.tests.model.factory.UserFactory.createUser;
+import static org.slf4j.LoggerFactory.getLogger;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * The TimeStamp test creates an experiment and bucket, assigns a user, and checks to see if the correct
+ * amount of impressions are being created when using different formats of timestamps at various times.
+ */
+public class TimeStampTest extends TestBase {
+    private static final Logger LOGGER = getLogger(BatchRuleTest.class);
+    public static final String COUNT_ERROR_MESSAGE = "Impression count does not match count of timestamps";
+    private final List<Experiment> validExperimentsLists = new ArrayList<>();
+    private Map<User, Assignment> assignments = new HashMap<>();
+
+    @DataProvider
+    public Object[][] sampleExperiment() {
+        String label = "timestampTest_" + System.currentTimeMillis();
+        Application application = new Application("LUA_timestamp");
+        String startTime = "2013-01-01T00:00:00+0000";
+        String endTime = TestUtils.relativeTimeString(1);
+        Double samplingPercent = 1.0;
+        Experiment experiment = new Experiment(label, application, startTime, endTime, samplingPercent);
+        return new Object[][]{
+                new Object[]{experiment}
+        };
+    }
+
+    @Test(dataProvider = "sampleExperiment")
+    public void setupExperiment(Experiment experimentData) {
+        LOGGER.debug(experimentData.toString());
```

Comment:
consider: add some textual context to the log statement

scenario: imagine you have a stack trace and you need to find in the code where it originated.

eg: LOGGER.debug(MessageFormat.format("posting experiments: %s", experimentData.toString())

note: i like to use MessageFormat.format() for this use case
note: further, double check the MessageFormat.format(...) supported templating syntax ... the above is psuedo code